src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -37,18 +38,26 @@ internal sealed class RegionDiscoveryProvider : IRegionDiscoveryProvider
             _imdsCallTimeoutMs = imdsCallTimeout;
         }
 
+        /*
+         * To be used only for tests
+         */
+        public void Clear()
+        {
+            s_region = null;
+        }
+
         public async Task<InstanceDiscoveryMetadataEntry> GetMetadataAsync(Uri authority, RequestContext requestContext)
         {
             ICoreLogger logger = requestContext.Logger;
             string environment = authority.Host;
-            InstanceDiscoveryMetadataEntry cachedEntry = _networkCacheMetadataProvider.GetMetadata(environment, logger);
+            Uri regionalizedAuthority = await BuildAuthorityWithRegionAsync(authority, requestContext).ConfigureAwait(false);
+            InstanceDiscoveryMetadataEntry cachedEntry = _networkCacheMetadataProvider.GetMetadata(regionalizedAuthority.Host, logger);
 
             if (cachedEntry == null)
             {
-                Uri regionalizedAuthority = await BuildAuthorityWithRegionAsync(authority, requestContext).ConfigureAwait(false);
                 CacheInstanceDiscoveryMetadata(CreateEntry(authority, regionalizedAuthority));
 
-                cachedEntry = _networkCacheMetadataProvider.GetMetadata(environment, logger);
+                cachedEntry = _networkCacheMetadataProvider.GetMetadata(regionalizedAuthority.Host, logger);
                 logger.Verbose($"[Region Discovery] Created metadata for the regional environment {environment} ? {cachedEntry != null}");
```

Comment:
>logger [](start = 16, length = 6)

nit: for all the `logger` instances, you can just use the rc directly `requestContext.Logger` and same w/ the methods, just pass in the rc.logger